flink-runtime/src/main/java/org/apache/flink/runtime/rest/RestClient.java

```diff
@@ -339,12 +338,26 @@ private static Request createRequest(String targetAddress, String targetUrl, Htt
 			.thenComposeAsync(
 				channel -> {
 					ClientHandler handler = channel.pipeline().get(ClientHandler.class);
-					CompletableFuture<JsonResponse> future = handler.getJsonFuture();
+
+					CompletableFuture<JsonResponse> future;
+					boolean success = false;
+
 					try {
-						httpRequest.writeTo(channel);
+						if (handler == null) {
```

Comment:
Do we know when/why this happens?

Reply:
No not really. It happened from time to time when running the `RestClientTest#testConnectionClosedHandling` in a loop. It might be a Netty bug but we should guard against this in order to harden the test case. I don't think that something bad happens if we fail the request with an `IOException` here.